### DISCUSSION TITLE
3d fragment index to 2d index

### DISCUSSION THREAD

**Discussion by orgw** [rdkit/rdkit#6386] [Q&A] [answered]
![image](https://github.com/rdkit/rdkit/assets/64889801/ef8aaf28-e641-4c17-9cd5-abf1a51fab17)
Hi i'm trying to highlight the fragments above into 2d. when converting to 2d, however, the atom index seems to differ. 
I have the 3d coordinates of the fragments. which i get the 3d atom index corresponding. how would i get the 2d coordinates of the fragments?

thanks always

**Comment by orgw** [comment]
one way would be to extract the fragments -> use substructsearch to find corresponding 2d coordinates. would there be another method?

**Reply by greglandrum** [reply]
Hi @orgw we can't answer your question without more information about what you're trying. Please share the code that you think is doing the wrong thing too.

**Comment by orgw** [comment]
```
pdb_ligand = os.path.join(casf_dir,lig_path[0]+'_ligand.sdf')
mol = Chem.SDMolSupplier(pdb_ligand)[0]
frags = Chem.GetMolFrags(BRICS.BreakBRICSBonds(mol),asMols=True)

frag_coord_list_tot = []

for frag in frags:
    frag_coord = frag.GetConformer().GetPositions()
    frag_coord = frag_coord - center_coord[:3].cpu().numpy()
    # index_per_frag.append([np.where(np.round(holo_coords.astype(np.float64),3) == np.round(coord.astype(np.float64),3))[0][0] for coord in frag_coord])
    frag_coord_list_tot.append(frag_coord)

conformer = mol.GetConformer()
atom_positions = conformer.GetPositions()
atom_positions = atom_positions - center_coord[:3].cpu().numpy()

mols = [mol] * len(frag_coord_list_tot)
mols2d = [Chem.MolFromSmiles(Chem.MolToSmiles(mol))] * len(frag_coord_list_tot)
target_atoms_total= []
for i in range(len(frag_coord_list_tot)):
    target_atoms = []
    for atom_idx, atom_position in enumerate(atom_positions):
        if atom_position in frag_coord_list_tot[i]:
            target_atoms.append(atom_idx)
    target_atoms_total.append(target_atoms)
Draw.MolsToGridImage(mols, highlightAtomLists=list(target_atoms_total))
```

![image](https://github.com/rdkit/rdkit/assets/64889801/bfb35593-61ec-4f82-80f6-95e7d3d37aa1)
![image](https://github.com/rdkit/rdkit/assets/64889801/fc8e7420-cced-4fc4-8308-3a53353120a9)

so the code above highlights the fragments. Now what i wanted to do was highlight the corresponding fragments in 2d( in 3d the fragments are correct) but when i convert the 3d molecules to 2d, the indexes change so i guess different parts are highlighted.

**Reply by greglandrum** [reply, marked as answer]
The atom indices change because you are converting to SMILES and back. If you want to get to molecules with 2D coordinates you can use:

from RDKit.Chem import rdDepictor

rdDepictor.Compute2DCoords(mol)

That will remove any existing coords and generate a set of 2d coordinates.

(Sorry I'm on my phone and can't format this properly)

**Reply by orgw** [reply]
thanks i will try
